tilelang/profiler/bench.py

```
@@ -98,3 +100,154 @@ def do_bench(
+
+# from https://github.com/deepseek-ai/DeepGEMM/blob/main/deep_gemm/testing/bench.py
+def do_bench_kineto(
+    fn: Callable,
+    kernel_names,
+    warmup: float = 25,
+    rep: float = 100,
+    _n_warmup: int = 0,
+    _n_repeat: int = 0,
+    fast_flush: bool = True,
+    return_mode: Literal["min", "max", "mean", "median"] = "mean",
+    trace_path: str = None,
```

Ruff (0.13.2)

PEP 484 prohibits implicit `Optional`

Convert to `Optional[T]`

(RUF013)